### DISCUSSION TITLE
Is it possible to hide some activity properties in dashboard?

### DISCUSSION THREAD

**Discussion by mlditga** [elsa-workflows/elsa-core#2655] [Q&A]
Hi,

I am wondering if there is a possibility to hide some activity properties in workflow instance view. We are trying to create a workflow by code starting with a custom HttpEndpoint. We store the body we receive with SetVariable and use it in the next activities. But some parts of that body are sensitive and should not be visible in the instance view. 
I saw that [issue](https://github.com/elsa-workflows/elsa-core/issues/1269#issuecomment-883473655) and tried to implement the notificationHandler. I removed the condition on the `descriptor.Type` but but it seems to work only for the custom HttpEndpoint (which is already customize to clear journal data). 
Should it work for all activities? Or is there another way to archive this?

Thank you very much!

**Comment by sfmskywalker** [comment]
You can customize what properties are displayed in the designer, but I don’t think there’s a way to hide it just for the workflow instance viewer without also hiding it from the definition editor.

Is that something you are looking for: a way to expose a property in the activity editor for the workflow editor user, but for it to remain hidden when viewing the workflow instance?

**Comment by mlditga** [comment]
Thank you for your answer. We want to create that workflow by code so I guess it should not be a problem if the property is not visible in the definition editor either. We would like to be able to use it in activities without the value (or whole property) being visible when viewing the workflow instance.

**Reply by sfmskywalker** [reply]
I see. Now that I think about it, I don't think there's currently a way to hide/redact content from the display Activity State (assuming this is what you're referring too):

![image](https://user-images.githubusercontent.com/938393/148545794-2810bb27-9d9a-4dc6-9cf8-0f254565223c.png)

But I can imagine you need control over what fields get displayed or redacted / masked.

If that's the case, I recommend you open a new issue with a feature request that we can discuss & design.

**Reply by mlditga** [reply]
Ok, I will do that. Thank you.
